### DISCUSSION TITLE
How to check the ssrContext in a Pinia store with Nuxt 3?

### DISCUSSION THREAD

**Discussion by eMeRiKa13** [vuejs/pinia#801] [Help and Questions] [answered]
Hi,

I would like to check the server side context in a Pinia Store with Nuxt 3.

I tried:
```
const ssrContext = this.$nuxt.ssrContext;
if (ssrContext) {
  // code
}
```

but the condition is always true.

In my template file, I use
```
const { ssrContext } = useNuxtApp();
if (ssrContext) {
  // code
}
```

ands it works properly.

Thank you

**Comment by posva** [comment]
if `this.$nuxt` doesn't work, it's probably a bug, likely on Nuxt (the source code is on packages/nuxt/src/templates/plugin.ts). You can create your own plugin:

```js
pinia.use(() => ({ $nuxtApp: useNuxtApp() }))
```

**Comment by eMeRiKa13** [comment, marked as answer]
Thank you for your answer `this.$nuxt` works.

The solution I was looking for is `if (process.server) { /* */ }`, don't need to use this.$nuxt.ssrContext here.
